src/main/scala/ai/privado/exporter/CollectionExporter.scala

```diff
@@ -77,13 +78,38 @@ class CollectionExporter(cpg: Cpg) {
         })
     })
 
+    collectionMethods.foreach(collectionMethod => {
+      collectionMethod.local
+        .and(_.tag.nameExact(Constants.id))
+        .foreach(localVar => {
+          try {
+            def addToMap(parameterId: String): Unit = {
+              if (!collectionLocalMapById.contains(parameterId))
+                collectionLocalMapById(parameterId) = ListBuffer()
+              collectionLocalMapById(parameterId).append(localVar)
+            }
+
+            localVar.tag
+              .nameExact(Constants.id)
+              .value
+              .filter(!_.startsWith(Constants.privadoDerived))
+              .foreach(addToMap)
+
+          } catch {
+            case e: Exception => logger.debug("Exception : ", e)
+          }
+        })
+    })
+
     val ruleInfo = ExporterUtility.getRuleInfoForExporting(collectionId)
     CollectionModel(
       collectionId,
       ruleInfo.name,
       ruleInfo.isSensitive,
       collectionParameterMapById
         .map(entrySet => processByParameterId(entrySet._1, entrySet._2.toList))
+        .toList ::: collectionLocalMapById
```

Comment:
what does the `:::` operator does?

Reply:
Its used for list concatenation.
Reference: https://alvinalexander.com/scala/how-merge-scala-lists-concatenate/